gramps_webapi/__main__.py

```diff
@@ -58,12 +59,71 @@ def cli(ctx, config):
 
 @cli.command("run")
 @click.option("-p", "--port", help="Port to use (default: 5000)", default=5000)
-@click.option("--tree", help="Tree ID", default=None)
+@click.option("-t", "--tree", help="Tree ID: '*' for multi-trees", default=None)
+@click.option(
+    "-o",
+    "--open",
+    help="Open gramps-web in browser: 'tab', 'window', or 'no'",
+    default="no",
+)
+@click.option(
+    "-d",
+    "--debug-level",
+    help="Debug level: 'info', 'debug', 'warning', 'critical'",
+    default="info",
+)
+@click.option("-l", "--log-file", help="Set logging file to this path", default=None)
+@click.option(
+    "--host", help="Set the host address for server to listen on", default="127.0.0.1"
+)
+@click.option(
+    "--max-workers",
+    help="Maximum number of workers for frontend; requires --use-wsgi",
+    default=None,
+)
+@click.option("--use-wsgi", is_flag=True, help="Add a wsgi wrapper around server")
 @click.pass_context
-def run(ctx, port, tree):
+def run(ctx, port, tree, host, open, debug_level, log_file, max_workers, use_wsgi):
     """Run the app."""
     app = ctx.obj["app"]
-    app.run(port=port, threaded=True)
+    debug_level = debug_level.upper()
+    if max_workers is None:
+        max_workers = min(32, os.cpu_count() + 4)
+
+    def open_webbrowser():
+        time.sleep(1.0)
+        new = {"tab": 0, "window": 1}[open]
+        webbrowser.open("http://%s:%s" % (host, port), new=0, autoraise=True)
+
+    if open != "no":
+        thread = Thread(target=open_webbrowser)
+        thread.start()
+
+    if log_file:
+        file_handler = logging.FileHandler(log_file, "w+")
+        app.logger.addHandler(file_handler)
+        app.logger.setLevel(debug_level)
+
+    print("Running gramps-web...")
```

Comment:
Gramps Web API

Reply:
Yes, I debated that. On one hand, the gramps-web-api is just the REST server. Thinking of suggesting renaming gramps-web to gramps-web-frontend,  plus gramps-web-api together as gramps-web (and I'm thinking about a package just called gramps-web). What do you think?

Reply:
> Thinking of suggesting renaming gramps-web to gramps-web-frontend, plus gramps-web-api together as gramps-web

That's exactly the naming we have. The frontend used to have a separate name (Gramps.js, "the Gramps Web frontend"), but we renamend the repository to `gramps-web` because it lead to confusion.

From the point of view of the combined web application, it would make sense to have a monorepo and combined packaging, but so far we still have them separated. Since the API can be used for other things as well, I think it kind of makes sense. And for proper web-based deployments, it's anyway more complicated (celery, nginx, etc.).

What we could do is to add a convenience command to download a frontend.

Reply:
I'd like to make a gramps-web-desktop that is pip installable, and runs from the command line with easy options, ready to go. The frontend could be included in repo, or downloaded. Which would you prefer?

Reply:
Changed the message to use gramps-web-api.

Reply:
> The frontend could be included in repo, or downloaded. Which would you prefer?

With the current development setup and release cycles, I think downloading is the only option as the frontend is released more frequently, but let's keep this for a separate PR.

By the way - in my opinion, a usable desktop-like Gramps Web needs a background task worker (but that's also a separate discussion).